### PR TITLE
[View] Dialog for Driver Path

### PR DESCRIPTION
**Pull request by maltewae**
**Needs #33 merged before!**

This pull request adds a widget to store driver paths in a general way. It has the same functionality as StageDriverSettingsDialog, except that DriverPathDialog is built in a general way.  Tests have been added to test functionality (e.g. button clicks) rather than appearance. 

**The code does not break any existing code. It simply adds code that is not already in use.**

## Example usage for `Stage3DSmarAct`
```python
def load_driver(cls, parent=None) -> bool:
    """
    Loads driver for SmarAct by open a dialog to specifiy the driver path.
    """
    driver_path_dialog = DriverPathDialog(
        parent,
        settings_file_path="mcsc_module_path.txt",
        title="Stage Driver Settings",
        label="SmarAct MCSControl driver module path",
        hint="Specify the directory where the module MCSControl_PythonWrapper is found.\nThis is external software,"
                 "provided by SmarAct GmbH and is available from them. See https://smaract.com."
    )
    parent.wait_window(driver_path_dialog)
    return driver_path_dialog.path_has_changed
```
Based on the return value  of `load_driver`, we can reload the modules without restarting LabExT.

### CODE REVIEW

**Review by marcoep — APPROVED** [review]
Good to merge, after #33 is merged.

**Comment by marcoep** [comment]
@maltewae pls rebase onto dev

**Comment by maltewae** [comment]
@marcoep Rebased

**Pull request merged**
